src/ZarrMsg.chpl

```diff
@@ -0,0 +1,125 @@
+module ZarrMsg {
+  use IO;
+  use ServerErrors, ServerConfig;
+  use FileIO;
+  use FileSystem;
+  use GenSymIO;
+  use List;
+  use Logging;
+  use Message;
+  use MultiTypeSymbolTable;
+  use MultiTypeSymEntry;
+  use NumPyDType;
+  use Sort;
+  use CommAggregation;
+  use AryUtil;
+  use CTypes;
+  use Zarr;
+  use ServerConfig;
+
+  use Reflection;
+  use ServerErrors;
+  use ServerErrorStrings;
+  use SegmentedString;
+
+  use Map;
+  use ArkoudaCTypesCompat;
+  use ArkoudaIOCompat;
+  use ArkoudaMathCompat;
+
+  private config const logLevel = ServerConfig.logLevel;
+  private config const logChannel = ServerConfig.logChannel;
+  const zarrLogger = new Logger(logLevel, logChannel);
+
+  @arkouda.registerND(cmd_prefix="readAllZarr")
+  proc readAllZarrMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
+    const storePath = msgArgs.getValueOf("store_path"), dtype = str2dtype(msgArgs.getValueOf("dtype"));
+    const rname = st.nextName();
+    select dtype {
+      when DType.Float64 {
+        var ar1 = readZarrArray(storePath, real, nd);
+        const a = makeDistArray(ar1);
+        st.addEntry(rname, createSymEntry(a));
```

Comment:
Am I understanding this correctly that we are reading into a DR (meaning all memory on locale 0) and then turning that into a distributed array?

Reply:
`readZarrArray` reads in the store into a distributed array using all locales; the `makeDistArray` call was added while debugging a comms issue, but I've fixed that elsewhere. These calls will be removed.